File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int i=0;
        int j=nums.size()-1;
        
        vector<int> res(2);
        
        while(i<=j){
            
            if(nums[i] + nums[j] == target){
                res[0]=i+1;
                res[1]=j+1;
                break;
            }
            else if(nums[i] + nums[j] > target){
                j--;
            }
            else{
                i++;
            }
        }
        
        return res;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        l, r = 0, len(numbers)-1
        res = []
        
        while l<r: 
            if numbers[l] + numbers[r] == target:
                res.append(l+1)
                res.append(r+1)
                break
            elif numbers[l] + numbers[r] > target:
                r -= 1
            else:
                l += 1
        return res